examples/energy-management-app/energy-management-common/src/ElectricalPowerMeasurementDelegate.cpp

```diff
@@ -0,0 +1,442 @@
+        .percentTypical = MakeOptional(static_cast<chip::Percent100ths>(500)),
+    },
+    // 2 - 5%, 3% Typ
+    {
+        .rangeMin       = 5'000,   // 5A
+        .rangeMax       = 100'000, // 100 A
+        .percentMax     = MakeOptional(static_cast<chip::Percent100ths>(5000)),
+        .percentMin     = MakeOptional(static_cast<chip::Percent100ths>(2000)),
+        .percentTypical = MakeOptional(static_cast<chip::Percent100ths>(3000)),
+    },
+};
+
+const MeasurementAccuracyRangeStruct::Type voltageAccuracyRanges[] = {
+    // 2 - 5%, 3% Typ
+    {
+        .rangeMin       = -500'000, // -500V
+        .rangeMax       = -100'000, // -100V
+        .percentMax     = MakeOptional(static_cast<chip::Percent100ths>(5000)),
+        .percentMin     = MakeOptional(static_cast<chip::Percent100ths>(2000)),
+        .percentTypical = MakeOptional(static_cast<chip::Percent100ths>(3000)),
+    },
+    // 0.1 - 1%, 0.5% Typ
+    {
+        .rangeMin       = -99'999, // -99.999V
+        .rangeMax       = 99'999,  //  99.999V
+        .percentMax     = MakeOptional(static_cast<chip::Percent100ths>(1000)),
+        .percentMin     = MakeOptional(static_cast<chip::Percent100ths>(100)),
+        .percentTypical = MakeOptional(static_cast<chip::Percent100ths>(500)),
+    },
+    // 2 - 5%, 3% Typ
+    {
+        .rangeMin       = 100'000, // 100 V
+        .rangeMax       = 500'000, // 500 V
+        .percentMax     = MakeOptional(static_cast<chip::Percent100ths>(5000)),
+        .percentMin     = MakeOptional(static_cast<chip::Percent100ths>(2000)),
+        .percentTypical = MakeOptional(static_cast<chip::Percent100ths>(3000)),
+    }
+};
+
+static const Structs::MeasurementAccuracyStruct::Type kMeasurementAccuracies[] = {
+    {
+        .measurementType  = MeasurementTypeEnum::kActivePower,
+        .measured         = true,
+        .minMeasuredValue = -50'000'000, // -50 kW
+        .maxMeasuredValue = 50'000'000,  //  50 kW
+        .accuracyRanges   = DataModel::List<const MeasurementAccuracyRangeStruct::Type>(activePowerAccuracyRanges),
+    },
+    {
+        .measurementType  = MeasurementTypeEnum::kActiveCurrent,
+        .measured         = true,
+        .minMeasuredValue = -100'000, // -100A
+        .maxMeasuredValue = 100'000,  //  100A
+        .accuracyRanges   = DataModel::List<const MeasurementAccuracyRangeStruct::Type>(activeCurrentAccuracyRanges),
+    },
+    {
+        .measurementType  = MeasurementTypeEnum::kVoltage,
+        .measured         = true,
+        .minMeasuredValue = -500'000, // -500V
+        .maxMeasuredValue = 500'000,  //  500V
+        .accuracyRanges   = DataModel::List<const MeasurementAccuracyRangeStruct::Type>(voltageAccuracyRanges),
+    },
+};
+
+uint8_t ElectricalPowerMeasurementDelegate::GetNumberOfMeasurementTypes()
+{
+    return ArraySize(kMeasurementAccuracies);
+};
+
+CHIP_ERROR ElectricalPowerMeasurementDelegate::GetAccuracyByIndex(uint8_t accuracyIndex,
```

Comment:
With this API shape, the assumption is that the AccuracyRanges for each of the structs stays in memory forever, right?

That's probably OK (though for bridges it might be a bit complicated....), but the delegate API should very clearly document this assumption.

Reply:
That works.